linux/src/file_linux.c

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "file.h"
+#include "file_internal.h"
+
+MU_DEFINE_ENUM_STRINGS(FILE_ASYNC_OPERATION, FILE_ASYNC_OPERATION_VALUES)
+
+typedef struct FILE_HANDLE_DATA_TAG
+{
+    int handle;
+    FILE_REPORT_FAULT user_report_fault_callback;
+    void* user_report_fault_context;
+}FILE_HANDLE_DATA;
+
+typedef struct FILE_LINUX_WRITE_TAG
+{
+    FILE_HANDLE_DATA handle;

```

Comment:
>FILE_HANDLE_DATA [](start = 4, length = 16)

why is the complete structure copied for every write? "why" questions... 

also seems to be copied for the read

Reply:
Thanks for catching that! That should be `FILE_HANDLE`. Changed in next iteration.

---
In reply to: [452462922](https://github.com/Azure/azure-c-pal/pull/7#discussion_r452462922) [](ancestors = 452462922)